fellow-submissions/devonte-duncan.md

```diff
@@ -0,0 +1,154 @@
+    that it has access to. Meanwhile "undefined" is logged because the function `testScope`
+    is referenced, however there is no return value within it, thus "undefined" is logged.
+
+**3. What will the following code log? Why?**
+    ```javascript
+    function hello() {
+      let a = 'hello';
+    }
+
+    hello();
+    console.log(a);
+    ```
+    This above code will log:
+        "undefined"
+        "ReferenceError: a is not defined"
+    
+    This is logged because the string variable `a` is declared only in the local scope 
+    of the `hello` function, thus trying to log `a` to the console will result in an error.
+    "undefined" is logged because the function `hello` is referenced, however there is no return 
+    value within it, thus "undefined" is logged.
+    
+
+**4. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    var a = 1;
+    ```
+    The above code will log:
+        "undefined"
+    
+    This logs "undefined" because even though the declaration of the variable `a` is hoisted
+    before runtime, while its initialization is not. Thus the value of `a` on line 60 is
+    logged as "undefined", until it assigned a value of 1 on line 62.
+    
+  
+**5. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    let a = 1;
+    ```
+    This logs an error, because `let` is a scoped variable, JavaScript cannot access it until 
+    it's been initialized. Therefore, even though it's declaration is hoisted, its initialization
+    must also occur before it's referenced in order to work.
+
+**6. In the code snippet below, the function declaration comes _after_ the variable declaration in the code below. Why does it still work?**
+    ```javascript
+    let name = "Devonte";
+
+    function sayItLoud() {
+      console.log(`${name}!!!`)
+    }
+
+    sayItLoud();
+    ```
+    The variable `name` was declared and initialized within the global scope, and 
+    therefore can be seen/accessed from all other scopes as well.
+    
+**7. Why does the following code throw an error? What is the key difference between it and the example above?**
+    ```javascript
+    sayItLoud();
+
+    function sayItLoud() {
+      console.log(`${greeting}!!!}`)
+    }
+
+    let greeting = "Hello"
+    ```
+    This logs an error, because `let` is a scoped variable, JavaScript cannot access it until 
+    it's been initialized. Therefore, even though it's declaration is hoisted and it's declared in 
+    the global scope, its initialization must also occur before it's referenced in order to work.
+
+**8. What does the following code log? Why?**
+    ```javascript
+    let problemSet;
+
+    function complete(assignment) {
+      assignment = "done";
+    }
+
+    complete(problemSet);
+    console.log(`Status for Problem Set: ${problemSet}`);
+    ```
+    
+    This code logs "Status for Problem Set: undefined". This is because the variable
```

Comment:
Answer is correct but explanation has a critical flaw. `problemSet` is not initialized in the body of that function. Remember: **function parameters become locally scoped variables in the body of the function**. Thus, when we assign the parameter a value in the function, it does not impact the binding of the argument passed to that function.